From 7fe13313fce14f380444c2a4530c5b5c4a3e390a Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Mon, 24 Jul 2017 11:20:09 +0800
Subject: [PATCH 2242/5242] MLK-16052-1 arm64: imx8qm: refine pcie power on
 imx8qm

commit  c6b154ac71306ef154411ed20eb061fb83fbc333 from
https://source.codeaurora.org/external/imx/linux-imx.git

- Refine the pd definitions of the imx8qm/qxp hsio.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi  |   34 +++++++++++++++++-------
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |   12 +++++++--
 2 files changed, 35 insertions(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index f45e37d..4a0fe49 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -484,20 +484,36 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-			pd_pcie0: PD_HSIO_PCIE_A {
-				reg = <SC_R_PCIE_A>;
-				#power-domain-cells = <0>;
-				power-domains =<&pd_hsio>;
-			};
-			pd_pcie1: PD_HSIO_PCIE_B {
-				reg = <SC_R_PCIE_B>;
+			pd_serdes0: PD_HSIO_SERDES_0 {
+				reg = <SC_R_SERDES_0>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_hsio>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				pd_pcie0: PD_HSIO_PCIE_A {
+					reg = <SC_R_PCIE_A>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_serdes0>;
+				};
+				pd_pcie1: PD_HSIO_PCIE_B {
+					reg = <SC_R_PCIE_B>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_serdes0>;
+				};
 			};
-			pd_sata0: PD_HSIO_SATA0 {
-				reg = <SC_R_SATA_0>;
+			pd_serdes1: PD_HSIO_SERDES_1 {
+				reg = <SC_R_SERDES_1>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_hsio>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				pd_sata0: PD_HSIO_SATA0 {
+					reg = <SC_R_SATA_0>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_serdes1>;
+				};
 			};
 			pd_gpio: PD_HSIO_GPIO {
 				reg = <SC_R_HSIO_GPIO>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 9bddb74..c9d29df 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -621,10 +621,18 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-			pd_pcie: PD_HSIO_PCIE_B {
-				reg = <SC_R_PCIE_B>;
+			pd_serdes1: PD_HSIO_SERDES_1 {
+				reg = <SC_R_SERDES_1>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_hsio>;
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				pd_pcie: PD_HSIO_PCIE_B {
+					reg = <SC_R_PCIE_B>;
+					#power-domain-cells = <0>;
+					power-domains =<&pd_serdes1>;
+				};
 			};
 		};
 
-- 
1.7.9.5

